tests/Doctrine/Tests/ORM/UnitOfWorkTest.php

```diff
@@ -707,6 +711,50 @@ public function testCanInstantiateInternalPhpClassSubclassFromUnserializedMetada
 
         self::assertInstanceOf(MyArrayObjectEntity::class, $this->unitOfWork->newInstance($classMetadata));
     }
+
+    public function testCallCommitInsideCommit() : void
+    {
+        $listener = new class()
+        {
+            public function preFlush(PreFlushEventArgs $eventArgs)
+            {
+                $eventArgs->getEntityManager()->getUnitOfWork()->commit();
+            }
+        };
+
+        $this->expectException(CommitInsideCommit::class);
+        $this->eventManager->addEventListener(Events::preFlush, $listener);
+        $this->unitOfWork->commit();
+    }
+
+    public function testCommitWithExceptionClearsInProgressFlag() : void
+    {
+        $listener = new class()
+        {
+            public $listenerCallCounter = 0;
+            public function preFlush(PreFlushEventArgs $eventArgs)
+            {
+                $this->listenerCallCounter++;
+                throw new RuntimeException();
+            }
+        };
+
+        $this->eventManager->addEventListener(Events::preFlush, $listener);
+        try {
+            $this->unitOfWork->commit();
+        } catch (RuntimeException $e) {
+        }
+
+        $this->eventManager->removeEventListener(Events::preFlush, $listener);
+
+        try {
+            $this->unitOfWork->commit();
+        } catch (CommitInsideCommit $e) {
+            $this->fail('UnitOfWork::$commitInProgress flag must be reset in case of exception during commit process');
+        }
+
+        $this->assertEquals(1, $listener->listenerCallCounter);
```

Comment:
Please use the static call like in the other tests and use `self::assertSame()` instead of `$this->assertEquals()`.

Reply:
`self::assertSame()`

Reply:
What is the difference in case when we are comparing scalars?

Reply:
`self::assertSame()` is stricter than `self::assertEquals()` and preferable. Similar to `===` compared with `==`.

Reply:
I know difference but there is no any for scalars.

As I see `assertEquals` is ok for scalars in existing code:
https://github.com/doctrine/orm/search?q=assertEquals&unscoped_q=assertEquals

Reply:
Scalar values should use `self::assertSame()` too, if there's no reason for a less stricter check.

Thank you for showing the other occurrences in existing tests. Those can be handled with a stricter check in another PR.